.github/workflows/post-merge-jobs.yml

```diff
@@ -20,8 +32,22 @@ permissions:
     contents: write
 
 jobs:
+    check_should_deploy:
```

Comment:
Note: we should do a test deploy to jobs after this merges to make sure this new config works. It should be fine since I copied it over to have parity with Public API and Internal API.